infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -0,0 +1,362 @@
+                            sessionsToRemove.add(session);
+                            metadataManager.removeSession(txn, session);
+                        }
+
+                        if(sinkClustersUnchanged.contains(session.getSinkClusterId())) {
+                            sessionsUnchanged.add(session);
+                            sessionToContextMap.get(session).setTopologyConfigId(topology.getTopologyConfigId());
+                            metadataManager.updateReplicationMetadataField(txn, session,
+                                    ReplicationMetadata.TOPOLOGYCONFIGID_FIELD_NUMBER, topology.getTopologyConfigId());
+                        }
+                    });
+                    txn.commit();
+                } catch (TransactionAbortedException e) {
+                    throw new RetryNeededException();
+                }
+
+                stopReplication(sessionsToRemove);
+                updateReplicationParameters(sessionsUnchanged);
+                createSessions();
+                return null;
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("Unrecoverable exception while removing session", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    private void updateReplicationParameters(Set<LogReplicationSession> sessionsUnchanged) {
+        if (replicationManager == null) {
+            return;
+        }
+        for (LogReplicationSession session : sessionsUnchanged) {
+            ClusterDescriptor cluster = topology.getSinkClusters().get(session.getSinkClusterId());
+            replicationManager.refreshRuntime(session, cluster, topology.getTopologyConfigId());
+        }
+    }
+
+    /**
+     * Create sessions (outgoing and incoming) along with metadata
+     *
+     */
+    private void createSessions() {
+        LogReplicationSession session;
+
+        try(TxnContext txn = corfuStore.txn(LogReplicationMetadataManager.NAMESPACE)) {
+            // TODO(V2): A replication context should be specific per session, as the set of streams to replicate is
+            // specific to a session (for now only supporting one default model) so using the same context
+            LogReplicationContext context = new LogReplicationContext(configManager,
+                    topology.getTopologyConfigId(), localCorfuEndpoint);
+
+            for(ClusterDescriptor sourceCluster : topology.getSourceClusters().values()) {
+                for(ClusterDescriptor sinkCluster : topology.getSinkClusters().values()) {
+
+                    // TODO(V2): for now only creating sessions for FULL TABLE replication model (assumed as default)
+                    session = LogReplicationSession.newBuilder()
+                            .setSourceClusterId(sourceCluster.clusterId)
+                            .setSinkClusterId(sinkCluster.clusterId)
+                            .setSubscriber(getDefaultSubscriber())
+                            .build();
```

Comment:
I think we need a way to recognize traditional FULL_TABLE replication session, maybe by extending `ClusterRole`?

because for new models like LOGICAL_GROUP and ROUTING_QUEUE, we can create their sessions by client registration. but FULL_TABLE is currently driven by topology (SOURCE/SINK). So if we pull in new models, the default FULL_TABLE session will always be created

@shama358 correct me if I'm wrong since I also didn't see your topologyDescriptor change includes that